pmd-core/src/main/java/net/sourceforge/pmd/PMD.java

```diff
@@ -60,7 +60,10 @@
  *
  * <p><strong>Warning:</strong> This class is not intended to be instantiated or subclassed. It will
  * be made final in PMD7.
+ *
+ * @deprecated This class is to be removed in PMD 7 in favor of a unified PmdCli entry point. {@link PmdAnalysis} should be used for non-CLI use-cases.
```

Comment:
This needs to be mentioned in the release notes in section "API Changes". And all the other deprecations as well.